pkg/kubelet/kubelet_node_status.go

```diff
@@ -515,13 +516,20 @@ func (kl *Kubelet) setNodeAddress(node *v1.Node) error {
 		} else {
 			var addrs []net.IP
 			addrs, err = net.LookupIP(node.Name)
+			if err != nil {
+				return fmt.Errorf("could not lookup node %s IP addresses using the local resolver. error: %v", node.Name, err)
+			}
 			for _, addr := range addrs {
-				if !addr.IsLoopback() && addr.To4() != nil {
+				err = kl.validateNodeIP()
```

Comment:
The review tool won't let me add a comment on an unchanged line, but I think it has bearing on the changeset, or at least I have questions...

It seems like the else clause on L516 is only called if kl.nodeIP is nil, right? That being the case, won't the if clause on L1015 of validateNodeIP() always succeed and return nil (IOW, skipping the checks, but not failing validation).

I'm having a hard time understanding why the test on L1015 is there. Can you elaborate?